### ISSUE TITLE
URL matches unexpectedly when multiple routes have a similar prefix

### ISSUE THREAD

**Issue by nwoltman** [delvedor/find-my-way#67]
#65 solved #59/#60, but it created this new bug:

```js
var router = require('find-my-way')()

router.on('GET', '/b/', function b () {})
router.on('GET', '/b/bulk', function b_bulk () {})
router.on('GET', '/b/ulk', function b_ulk () {})

console.log(router.find('GET', '/bulk')) // { handler: [Function: b_ulk], ... }
```

#65 may not have been the right fix. It seems like something is going on with how prefixes are matched and might have something to do with the `label` property on nodes.

**Comment by DenisFrezzato** [comment]
I found a similar erroneous behaviour with parametric routes. 

```
router.on('GET', '/foo/:fooParam', function foo() {})
router.on('GET', '/foo/bar/:barParam', function fooBar() {})
router.on('GET', '/foo/search', function fooSearch() {})
router.on('GET', '/foo/submit', function fooSubmit() {})

console.log(router.find('GET', '/foo/awesome-parameter')) // { handler: [Function: foo]
console.log(router.find('GET', '/foo/b-first-character')) // null
console.log(router.find('GET', '/foo/s-first-character')) // { handler: [Function: foo]
console.log(router.find('GET', '/foo/se-prefix')) // null
```

**Comment by mcollina** [comment]
@DenisFrezzato would you like to send a PR?

**Comment by nwoltman** [comment]
@DenisFrezzato Thanks for the extra test cases! That helped me figure out the fix in #70 :)

**Comment by DenisFrezzato** [comment]
@nwoltman thank you for the fix! I have tried to fix by myself, but I couldn't find out a nice solution :smile:
